### ISSUE TITLE
Cant pip install your package

### ISSUE THREAD

**Issue by MehdiBela** [valdergallo/django-compress-field#7]
Hello,

I want to try your package for my project, but when I try to pip install it, I get an error. Below is full traceback. 

pip install django-compress-field
Collecting django-compress-field
  Downloading django-compress-field-0.9.8.tar.gz
```
Exception:
Traceback (most recent call last):
  File "c:\program files (x86)\python\lib\site-packages\pip\compat\__init__.py", line 73, in console_to_str
    return s.decode(sys.__stdout__.encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x82 in position 27: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\program files (x86)\python\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "c:\program files (x86)\python\lib\site-packages\pip\commands\install.py", line 324, in run
    requirement_set.prepare_files(finder)
  File "c:\program files (x86)\python\lib\site-packages\pip\req\req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "c:\program files (x86)\python\lib\site-packages\pip\req\req_set.py", line 634, in _prepare_file
    abstract_dist.prep_for_dist()
  File "c:\program files (x86)\python\lib\site-packages\pip\req\req_set.py", line 129, in prep_for_dist
    self.req_to_install.run_egg_info()
  File "c:\program files (x86)\python\lib\site-packages\pip\req\req_install.py", line 439, in run_egg_info
    command_desc='python setup.py egg_info')
  File "c:\program files (x86)\python\lib\site-packages\pip\utils\__init__.py", line 676, in call_subprocess
    line = console_to_str(proc.stdout.readline())
  File "c:\program files (x86)\python\lib\site-packages\pip\compat\__init__.py", line 75, in console_to_str
    return s.decode('utf_8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x82 in position 27: invalid start byte
```

As my dev computer isn't supposed to be connected to internet, I'd rather not try to clone it from git. Could you please have a look and let me know. Btw, I'm using Python 3.6.1 and Django 1.11 if that can be of any help.

**Comment by MehdiBela** [comment]
I've tried git installation from my own PC and it works fine, only pip install isn't working.

**Comment by MehdiBela** [comment]
BTW I think there's also an issue with your setup.py, I had to manually move the "compress_field" folder to site_packages to be able to import it.

Another issue I have is the setting "FILE_COMPRESS_DELETE_OLD_FILE" = True which has no effect (both file and zipped file are saved).

Regards.

**Comment by valdergallo** [comment]
Wowww ... is the first time that somebody use this project in python 3.6 with this django. 
I need make some testes e fix the problems ... thx for report. 

I know that you can install with pip + github rep 

```
pip install git+https://github.com/valdergallo/django-compress-field.git
``

**Comment by MehdiBela** [comment]
That's probably to Python version then indeed. I cannot downgrade the version I use for my project however. Please let me know about your tests results. Oh and by the way, great job you did there though.

**Comment by valdergallo** [comment]
Its done !!! 

I tested the `FILE_COMPRESS_DELETE_OLD_FILE` and it's working. Im using the the `storage.delete` command to delete the files. Check if your site has permission to manager the files. 

And I removed some strings with unicode ... 

``` 
(env) valder@mark5:/opt/loggi/okok $ pip install django-compress-field
Collecting django-compress-field
  Downloading django_compress_field-0.10.0-py2.py3-none-any.whl
Collecting django>=1.2 (from django-compress-field)
  Using cached Django-1.11.5-py2.py3-none-any.whl
Collecting pytz (from django>=1.2->django-compress-field)
  Using cached pytz-2017.2-py2.py3-none-any.whl
Installing collected packages: pytz, django, django-compress-field
Successfully installed django-1.11.5 django-compress-field-0.10.0 pytz-2017.2
(env) valder@mark5:/opt/loggi/okok $ pip freeze
Django==1.11.5
django-compress-field==0.10.0
pytz==2017.2
```

And all its working :D 

Thx again for report

**Comment by MehdiBela** [comment]
Hi again and thanks for you work !! I'm off till monday but I'll try it asap. 
PS : 
I've just tried quickly a pip install django-compress-field, it raises not error, however "compress_field" isn't recognized in Django settings file. 
Contrary to you I'm not working with a virtualenv that's might be the reason.
I'll try it more thoroughly on Monday and let you know.

**Comment by MehdiBela** [comment]
Hello,

So I've been able to make a few tests and here is a report of the issues I've got:

1. pip install isn't working for me : (I'm using Windows if that might help) I just get the .info folder but not the package.

2. If I clone the package from git and then setup.py install, "compress_field" folder isn't moved to site-packages, I have to do it manually to be able to set it in Django installed apps.

3. So far when using the .compress() method, it doesn't store it in the right media folder. Here are the things I tried : 

- From shell, I use the compress method on a an object stored in database : my object has a lambda file stored in MyProject/media/soft_sources. When I do my_object.file.compress() , the zipped file is stored in MyProject/media 

- In my code, I have a signal handler to generate the stored file checksum like this : 

```
@receiver(post_save)
def create_checksum(sender, instance, **kwargs):
    try:
        if instance.source:
            qset = sender.objects.filter(id=instance.id)
            hash_md5 = hashlib.md5()
            try:
                instance.source.compress()
                qset.update(source=instance.source)
                with open(qset[0].source.path, "rb") as f:
                    for chunk in iter(lambda: f.read(4096), b""): #for large files
                        hash_md5.update(chunk)
                qset.update(signature=hash_md5.hexdigest())
            except FileNotFoundError:
                pass
```
With this, queryset is correctly updated but the zipped file isnt stored, or I dont know where.  Here is all the info I've got, if you could have a look I would be very glad. Let me know if I can give you more information.

**Comment by valdergallo** [comment]
Wow ... 

1. pip install isn't working for me : (I'm using Windows if that might help) I just get the .info folder but not the package.
```
Are you using python3.6 in windows ??  Check this packages (https://pypi.python.org/pypi/django-compress-field/0.10.0). I cant compile this project for windows ... setup.py bdist_wininst not will works for me .... Im using OSX ... 

It is not a bug  ... 
```

2. If I clone the package from git and then setup.py install, "compress_field" folder isn't moved to site-packages, I have to do it manually to be able to set it in Django installed apps.

```
I don`t  undestand why this is happing in your computer ... Are you using command python setup.py build install ?
```

3. So far when using the .compress() method, it doesn't store it in the right media folder. Here are the things I tried :

```
YES ... Windows have one different filemanager ... Im using the django.file.storege to move the files. 
``

Why you dont try use vagrant or docker to developer in Windows ??

https://www.vagrantup.com/docs/installation/
https://realpython.com/blog/python/django-development-with-docker-compose-and-machine/

**Comment by MehdiBela** [comment]
1. I've tried pip install with a Kali Linux computer and got same result : only .info folder

2. Again I tried cloning from git and doing both setup.py install and setup.py build install and again got same result : I had to manually move compress_field from django-compress-field folder. 

3. 

> YES ... Windows have one different filemanager

I dont know if that's related actually. However in your package description it's written "Windows Support"

> Why you dont try use vagrant or docker to developer in Windows ?? 

As I told you I'm supposed to have restricted access on my dev computer and I'm not supposed to install extra software. Anyway, we dont know yet which OS we will use for deployment server and I'd rather have a cross-platform solution.
